Policies/Kubernetes/enforce-apparmor-profile/rego/src_test.rego

```diff
@@ -0,0 +1,111 @@
+package k8sazureenforceapparmor
+
+test_input_apparmor_allowed_empty {
+    input := { "review": input_review_container, "parameters": input_parameters_empty}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_apparmor_not_allowed_no_annotation {
+    input := { "review": input_review_no_annotation, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_apparmor_container_allowed_in_list {
+    input := { "review": input_review_container, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_apparmor_container_not_allowed_not_in_list {
+    input := { "review": input_review_container, "parameters": input_parameters_not_in_list}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_apparmor_containers_allowed_in_list {
+    input := { "review": input_review_containers, "parameters": input_parameters_in_list}
```

Comment:
In test input_review_no_annotation and input_parameters_in_list, a violation is flagged.
In test input_review_containers and input_parameters_in_list, violation is not flagged even though one of the containers (nginx) doesn't have a annotation. 
Aren't both cases mismatch? Just one of the containers having annotation is good enough?
Also is annotation a must? For the pods that do not have annotation, we flag them as violated?

Reply:
The pod has an annotation, rather than the container. This annotation is the one that matters, so this policy throws a pod level violation rather than a container level.

Yes, the annotation is necessary, if there's no annotation, it's flagged as a violation as the annotation is counted as null, and that doesn't match any of the values in the `allowedProfiles` list

The missing feature here, and for privilege escalation and probably a couple of others is that pod security policies allow modification, so they all have the ability to set a default, and that default is applied before the pod is created. Gatekeeper does not have modification abilities at this time, so all of these flags need to be added manually.

Reply:
The issue is even though only nginx2 container has app armor annotation, violation is not flagged. As discussed offline, please update the rego logic to consider this case and add tests.